src/sourcekit-lsp/LanguageClientManager.ts

```diff
@@ -609,15 +619,27 @@ export class LanguageClientManager {
         this.clientReadyPromise = client
             .start()
             .then(() => {
+                // Now that we've started up correctly, start the error handler to auto-restart
+                // if sourcekit-lsp crashes during normal operation.
+                errorHandler.enable();
+
                 if (this.workspaceContext.swiftVersion.isLessThan(new Version(5, 7, 0))) {
                     this.legacyInlayHints = activateLegacyInlayHints(client);
                 }
 
                 this.peekDocuments = activatePeekDocuments(client);
             })
             .catch(reason => {
-                this.workspaceContext.outputChannel.log(`${reason}`);
-                // if language client failed to initialise then shutdown and set to undefined
+                const lspOutputChannel = client.outputChannel as SwiftOutputChannel;
```

Comment:
There seems to be some weird timing error where this code doesn't seem to be called when you pass in a bad command line parameter to SKLSP, unless you stick breakpoints in here.

Reply:
I just reverted this block, it doesn't actually get called when sourcekit-lsp fails to launch. I think its only if there is some error on the VS Code side.